src/swap/useSwapQuote.ts

```diff
@@ -55,7 +55,7 @@ async function createBaseSwapTransactions(
 
   // If the sell token is ERC-20, we need to check the allowance and add an
   // approval transaction if necessary
-  if (allowanceTarget !== zeroAddress && fromToken.address) {
+  if (allowanceTarget !== zeroAddress && !fromToken.isNative) {
```

Comment:
The approval can be needed for CELO tokens.
It depends what the router does.

The approval is not needed for CELO when the `value` field is not 0.
But we can/should control that in our swap API instead.

Reply:
Yes! I think we should return the `zeroAddress` in the allowance target when the value field is different than 0 in our backend.

Reply:
Just to be clear, this change will make all non-native Celo swaps fail, since it will be missing the approval for it.